pages/using_react.md

```diff
@@ -0,0 +1,182 @@
+  ---
+layout: default
+title: Using React
+permalink: /using-react/
+sitemap:
+  - /using_react.html
+    priority: 0.7
+    lastmod: 2015-01-29T23:41:00-00:00
+---
+
+# <i class="fa fa-html5"></i> Using React (with Redux)
+This section refers to the JavaScript library **React** used with **Redux**.
+## Project Structure
+
+The JHipster client code can be found under `src/main/webapp`, and follows closely the [Piotr Witek React style guide](https://github.com/piotrwitek/react-redux-typescript-guide/blob/master/README.md).
+ Please read this guide first if you have any question on our application structure, file names, TypeScript conventions...
+
+For React routes we follow a dash cased naming convention so that the URLs are clean and consistent.
+When you generate an entity the route names, route URLs and REST API endpoint URLs are generated according to this convention, also entity names are automatically pluralized where required.
+
+Here is the main project structure:
+[comment] <> (TODO: what is the purpose of typings.d.ts)
```

Comment:
This is where you will define custom type module definitions, for example, handle json files, modify type of a lib class etc.

Reply:
I'll update these